### ISSUE TITLE
A bug of torch.nn.functional.grid_sample(padding_mode)

### ISSUE THREAD

**Issue by HanDarkholme** [pytorch/pytorch#19717]
## 🐛 Bug

`torch.nn.functional.grid_sample(input, grid, mode='bilinear', padding_mode='zeros')`,  
 
the bug is conerned on the **padding mode**. By default, `padding_mode='zeros'` means when values of the grid are out of [-1,1], then those values of the grid are setting to be 0. However, this default padding mode changes the values of **the input** when values of the corresponding position in the grid are out of [-1,1]. 
## To Reproduce 

1. Prepare a sample image A(make sure its minimum value of pixels are greater than 0), generate a tensor of the grid(making some values out of [-1,1], the same size of A).
1. Use `torch.nn.functional.grid_sample` to warp A by the generated grid(use `padding_mode='zeros'`) to achieve B.
1. print the maximum and the minimum value of the pixels in A and B.  

## Expected behavior

Then the minimum value of B is 0, which is incorrect.
## Environment
 - PyTorch Version : 1.0
 - OS: linux
 - How you installed PyTorch (`conda`, `pip`, source): conda
 - Python version: 3.6.4
 - CUDA/cuDNN version: 10.0
 - GPU models and configuration:

**Comment by SsnL** [comment]
The behavior is correct. It doesn't change the value of the grid, but it specifies what output should be put in B when the grid is out of border.

**Comment by HanDarkholme** [comment]
Sorry for this. But I think in the official document of pytorch , the illustration of this function seems not so clear about what you've mentioned, also may due to my poor english, I misunderstood this function.

**Comment by SsnL** [comment]
closed via https://github.com/pytorch/pytorch/pull/19754
